spec/services/feeds/import_spec.rb

```diff
@@ -208,4 +208,15 @@
       expect(body_markdown).to include(expected_image_markdown)
     end
   end
+
+  context "when multiple users fetch from the same feed_url" do
+    it "fetches the articles in both accounts (if feed_mark_canonical = false)" do
+      rss_feed_user1 = create(:user, feed_url: link)
+      rss_feed_user2 = create(:user, feed_url: link)
+
+      expect do
+        described_class.call(users: User.where(feed_url: link))
+      end.to change(rss_feed_user1.articles, :count).by(10) and change(rss_feed_user2.articles, :count).by(10)
```

Comment:
We should use RSpec's `and` DSL method over the `and` keyword:

```ruby
expect { described_class.call(users: User.where(feed_url: link)) }
  .to change(rss_feed_user1.articles, :count).by(10)
  .and change(rss_feed_user2.articles, :count).by(10)
```

The two do not behave the same way here:

1. With the `and` keyword the spec succeeds even if the second assertion is wrong:  
    ![image](https://user-images.githubusercontent.com/47985/125018030-76f2a480-e09e-11eb-82b6-db02603946a1.png)
2. With the `and` DSL method the results are correctly combined:
    ![image](https://user-images.githubusercontent.com/47985/125017952-5296c800-e09e-11eb-8353-8e558578233b.png)

That's because the `and` keyword uses truthiness, just like the `&&` operator. The `change(),by()` expression returns an instance of `RSpec::Matchers::BuiltIn::ChangeRelatively` which is neither `nil` nor `false`. The `and` DSL method looks at the result of the assertion instead.

Reply:
Thanks for the great explanation! I was concerning to include `and` with DSL methods, but I was unaware of the DSL `.and` method and the differences. This is really insightful!